packages/kbn-alerts-ui-shared/src/alerts_search_bar/index.test.tsx

```diff
@@ -0,0 +1,173 @@
+  </button>
+));
+
+const toastsMock = { toasts: { addWarning: jest.fn() } } as unknown as ToastsStart;
+const httpMock = {
+  post: jest.fn(),
+} as unknown as HttpStart;
+
+describe('AlertsSearchBar', () => {
+  beforeEach(() => {});
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders correctly', async () => {
+    render(
+      <AlertsSearchBar
+        rangeFrom="now/d"
+        rangeTo="now/d"
+        query=""
+        onQuerySubmit={jest.fn()}
+        onFiltersUpdated={jest.fn()}
+        appName={'test'}
+        featureIds={['observability', 'stackAlerts']}
+        unifiedSearchBar={unifiedSearchBarMock}
+        toasts={toastsMock}
+        http={httpMock}
+        dataService={mockDataPlugin}
+      />
+    );
+    expect(await screen.findByTestId('querySubmitButton')).toBeInTheDocument();
+  });
+
+  it('calls onQuerySubmit correctly', async () => {
+    const onQuerySubmitMock = jest.fn();
+
+    render(
+      <AlertsSearchBar
+        rangeFrom="now/d"
+        rangeTo="now/d"
+        query=""
+        onQuerySubmit={onQuerySubmitMock}
+        onFiltersUpdated={jest.fn()}
+        unifiedSearchBar={unifiedSearchBarMock}
+        toasts={toastsMock}
+        http={httpMock}
+        dataService={mockDataPlugin}
+        appName={'test'}
+        featureIds={['observability', 'stackAlerts']}
+      />
+    );
+
+    fireEvent.click(await screen.findByTestId('querySubmitButton'));
+
+    await waitFor(() => {
+      expect(onQuerySubmitMock).toHaveBeenCalled();
+    });
+  });
+
+  it('calls onFiltersUpdated correctly', async () => {
+    const onFiltersUpdatedMock = jest.fn();
+    const filters = [
+      {
+        meta: {
+          negate: false,
+          alias: null,
+          disabled: false,
+          type: 'custom',
+          key: 'query',
+        },
+        query: { bool: { filter: [{ term: { 'kibana.alert.rule.consumer': 'stackAlerts' } }] } },
+        $state: { store: 'appState' },
+      },
+    ] as Filter[];
+
+    const newUnifiedSearchBarMock = jest.fn().mockImplementation((props) => (
+      <button
+        data-test-subj="filtersSubmitButton"
+        onClick={() => props.onFiltersUpdated(filters)}
+        type="button"
+      >
+        {'Hello world'}
+      </button>
+    ));
+
+    render(
+      <AlertsSearchBar
+        rangeFrom="now/d"
+        rangeTo="now/d"
+        query=""
+        onQuerySubmit={jest.fn()}
+        onFiltersUpdated={onFiltersUpdatedMock}
+        unifiedSearchBar={newUnifiedSearchBarMock}
+        toasts={toastsMock}
+        http={httpMock}
+        dataService={mockDataPlugin}
+        appName={'test'}
+        featureIds={['observability', 'stackAlerts']}
+      />
+    );
+
+    fireEvent.click(await screen.findByTestId('filtersSubmitButton'));
+
+    await waitFor(() => {
+      expect(onFiltersUpdatedMock).toHaveBeenCalledWith(filters);
+      expect(mockDataPlugin.query.filterManager.setFilters).toHaveBeenCalledWith(filters);
```

Comment:
I'm not sure about these ones. Yeah, we are checking that it's calling this function and we know that for now it means that the badge is rendered as expected. But if something in that fn breaks how we render the badge we won't notice that it's actually broken.

Reply:
hmm.. good point. Maybe adding a functional test to verify badge is rendered correctly would help. What do you think?

Reply:
that would be perfect 🚀

Reply:
[added](https://github.com/elastic/kibana/pull/193623/commits/4a5d01e75f7879721296ecd4c1266f8f33e09006)